File: q1624/q1624.java
class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        int maxLen = -1;
        Map<Character, Integer> indexMap = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            char currChar = s.charAt(i);
            if (indexMap.containsKey(currChar)) {
                maxLen = Math.max(maxLen, i - indexMap.get(currChar) - 1);
            } else {
                indexMap.put(currChar, i);
            }
        }
        return maxLen;
    }
}

File: q1624/q1624.cpp
class Solution
{
public:
    int maxLengthBetweenEqualCharacters(string s)
    {
        unordered_map<char, int> indexMap;
        int maxLen = -1;
        for (int i = 0; i < s.size(); i++)
        {
            if (indexMap.find(s[i]) != indexMap.end())
            {
                maxLen = max(maxLen, i - indexMap[s[i]] - 1);
            }
            else
            {
                indexMap[s[i]] = i;
            }
        }
        return maxLen;
    }
};